### ISSUE TITLE
SECOND not detected as a MySQL keyword

### ISSUE THREAD

**Issue by jetexe** [sqlfluff/sqlfluff#5854]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

On SQL query:

```SQL
SELECT
  t.id,
  TIMESTAMPDIFF(SECOND, m.time_created, m.time_updated) AS s,
  TIMESTAMPDIFF(MINUTE, m.time_created, m.time_updated) AS m,
  TIMESTAMPDIFF(HOUR, m.time_created, m.time_updated) AS h,
  TIMESTAMPDIFF(DAY, m.time_created, m.time_updated) AS d,
  TIMESTAMPDIFF(MONTH, m.time_created, m.time_updated) AS mon,
  TIMESTAMPDIFF(YEAR, m.time_created, m.time_updated) AS y
FROM table1 AS m
LEFT JOIN table2 AS t
  ON m.id = t.id;
```

Got an error: 
```
L:   3 | P:  17 | RF02 | Unqualified reference 'SECOND' found in select with more than one referenced table/view.
                       | [references.qualification]
L:   4 | P:  17 | RF02 | Unqualified reference 'MINUTE' found in select with more than one referenced table/view.
                       | [references.qualification]
L:   5 | P:  17 | RF02 | Unqualified reference 'HOUR' found in select with more than one referenced table/view.
                       | [references.qualification]
L:   6 | P:  17 | RF02 | Unqualified reference 'DAY' found in select with more than one referenced table/view.
                       | [references.qualification]
L:   7 | P:  17 | RF02 | Unqualified reference 'MONTH' found in select with more than one referenced table/view.
                       | [references.qualification]
L:   8 | P:  17 | RF02 | Unqualified reference 'YEAR' found in select with more than one referenced table/view.
                       | [references.qualification]
```

### Expected Behaviour

No error

### Observed Behaviour

`TIMESTAMPDIFF` [doccumentation](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_timestampdiff)

### How to reproduce

```SQL
SELECT
  t.id,
  TIMESTAMPDIFF(SECOND, m.time_created, m.time_updated) AS s,
  TIMESTAMPDIFF(MINUTE, m.time_created, m.time_updated) AS m,
  TIMESTAMPDIFF(HOUR, m.time_created, m.time_updated) AS h,
  TIMESTAMPDIFF(DAY, m.time_created, m.time_updated) AS d,
  TIMESTAMPDIFF(MONTH, m.time_created, m.time_updated) AS mon,
  TIMESTAMPDIFF(YEAR, m.time_created, m.time_updated) AS y
FROM table1 AS m
LEFT JOIN table2 AS t
  ON m.id = t.id;
```

### Dialect

`MySQL`

### Version

`3.0.5`

### Configuration

```
[sqlfluff]
# Below controls SQLFluff output, see max_line_length for SQL output
output_line_length = 120
# Set to zero or negative to disable checks
max_line_length = 200
# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
dialect = mysql
# Comma separated list of rules to check, default to all
rules = all
# Ignore errors by category (one or more of the following, separated by commas: lexing,linting,parsing,templating)
ignore = parsing, templating
# Comma separated list of rules to exclude, or None
exclude_rules = capitalisation.identifiers,capitalisation.functions,layout.spacing,layout.indent
# Warn only for rule codes (one of more rule codes, seperated by commas: e.g. LT01,LT02)
warnings = LT01,CP02,LT02
# Whether to warn about unneeded '-- noqa:' comments
warn_unused_ignores = True
# CPU processes to use while linting. 0 means all cpus
processes = 0

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
